cmd/frostfs-node/main.go

```diff
@@ -96,14 +95,13 @@ func initApp(c *cfg) {
 	initAndLog(c, "reputation", initReputationService)
 	initAndLog(c, "notification", initNotifications)
 	initAndLog(c, "object", initObjectService)
-	initAndLog(c, "pprof", initProfiler)
-	initAndLog(c, "prometheus", initMetrics)
 	initAndLog(c, "tree", initTreeService)
 	initAndLog(c, "control", initControlService)
 
 	initAndLog(c, "morph notifications", listenMorphNotifications)
 
 	c.workers = append(c.workers, newWorkerFromFunc(c.configWatcher))
+	c.workers = append(c.workers, newWorkerFromFunc(c.terminateWatcher))
```

Comment:
what if initialization is taking too long but no os interuption signals are listened yet? mb that is not critical at all
/cc @fyrchik

Reply:
@carpawell see your point. in this case setting of the health status should be under the mutex and we need to exit once the signal trapped and status not equal READY, right?

Reply:
yes, two things bother me:

1. signal listening moved in the end of the init stage in that PR;
2. component init could take time and context cancel does not seem to handle that

Reply:
@carpawell right, it is impossible to cover that with context, when health state is undefined or starting. I think here we need to decide set o services for graceful shutdown, local storage is the first candidate, try close them and exit. @fyrchik your thoughts on this?

Reply:
i think we can just leave an issue for that and merge that PR

Reply:
Could you elaborate, what is the problem, exactly?
Simple mutex in `setHealthStatus` seems easy to implement if needed.

Reply:
What should we do if we stuck at local storage initialization or at service startup - should we try to shutdown normally or we need to try to close limited set of workers and perform `os.Exit(1)`?

Reply:
Let's wait for now -- it is a feature, this task is about shutdown.
Ideally we would accept context for all operations with storage engine.

Reply:
Moved to the top of the `initApp()`. Created new issue #49 to cover this case.